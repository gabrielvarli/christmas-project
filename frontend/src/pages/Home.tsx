import Hero from "../components/Hero";
import { getHeroPropsForPage } from "../components/Hero";
import { HeroProps } from "../models/productModels";
import Countdown from "../components/Countdown";
import { Link } from "react-router-dom";

const Home = () => {
  const heroInfo = getHeroPropsForPage("Home") as HeroProps;

  return (
    <main>
      {/* Hero Section */}
      <Hero {...heroInfo} />

      {/* Countdown Section */}
      <section className="w-full bg-rose-400 text-white text-center flex flex-col sm:flex-row items-center justify-center py-8">
        <div className="flex flex-col items-center sm:items-start mx-8">
          <div className="relative">
            <h1 className="font-dancing font-bold text-4xl">Nedr√§kning till</h1>
            <img
              src="/images/santaclausremovebg.png"
              alt="Christmas Santa Claus"
              className="hidden md:block absolute -top-11 -left-12 w-20 transform scale-x-[-1] rotate-[-25deg]"
            />
          </div>
          <h2 className="font-dancing font-bold text-3xl">Juldagen</h2>
        </div>
        <div className="mt-6 sm:mt-0 mx-4 flex justify-center items-center">
          <Countdown />
        </div>
      </section>

      {/* Highlighted Products Section */}
      <section className="my-16 w-full px-6 sm:px-12 lg:px-20 flex flex-wrap gap-8 justify-evenly">
        {[
          {
            image: "/images/julgranframsida.png",
            alt: "Julgran framsida",
            title: "Klassiska Julgranar",
            link: "/products",
          },
          {
            image: "/images/belysning.png",
            alt: "Belysning",
            title: "Belysning",
            link: "/products",
          },
        ].map((item, index) => (
          <article key={index} className="relative group">
            <Link to={item.link}>
              <img
                src={item.image}
                alt={item.alt}
                className="w-72 xs:w-96 h-80 xs:h-[450px] rounded-lg border-4 border-black cursor-pointer transition-transform duration-300 group-hover:scale-105 group-hover:shadow-lg"
              />
            </Link>
            <h3 className="text-4xl text-black absolute bottom-4 left-3 transition-colors duration-300 group-hover:text-rose-600">
              {item.title}
            </h3>
          </article>
        ))}
      </section>
    </main>
  );
};

export default Home;
