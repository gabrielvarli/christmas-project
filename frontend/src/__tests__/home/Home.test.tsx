import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "../../pages/Home";
import { MemoryRouter } from "react-router-dom";

describe("Home Page", () => {
  let container: HTMLElement;

  beforeEach(() => {
    const renderResult = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    container = renderResult.container;
  });

  it("renders a call-to-action link with text 'Visit Shop'", () => {
    const link = screen.getByRole("link", { name: /Visit Shop/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "/products"); // Kontrollera att länken har rätt URL
  });

  it("renders a call-to-action link with 'About Us'", () => {
    const link = screen.getByRole("link", { name: /About us/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "/about"); // Kontrollera att länken har rätt URL
  });

  it("renders the hero section with the correct title", () => {
    const heading = screen.getByRole("heading", {
      name: /Skapa en magisk Julstämning/i,
    });
    expect(heading).toBeInTheDocument();
  });

  it("renders an image of a Christmas decoration", () => {
    const imgElement = screen.getByRole("img", {
      name: /christmas santa claus/i,
    });
    expect(imgElement).toHaveAttribute("src", "/images/santaclausremovebg.png");
  });

  it("renders countdown timer", () => {
    const countdown = screen.getByText(/Days/i);
    expect(countdown).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});
