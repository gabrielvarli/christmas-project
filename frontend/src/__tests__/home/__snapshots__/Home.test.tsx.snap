// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches the snapshot 1`] = `
<div>
  <main>
    <section
      class="relative bg-[url('/images/framsidajulgran.png')] bg-cover bg-center bg-no-repeat max-h-[509px] flex items-center justify-center"
    >
      <div
        class="absolute inset-0 bg-gradient-to-b from-transparent via-black/30 to-black/50 z-0"
      />
      <div
        class="relative z-10 mx-auto max-w-screen-xl px-6 py-24 text-white"
      >
        <div
          class="text-center sm:text-left"
        >
          <h1
            class="text-4xl sm:text-5xl lg:text-6xl font-bold font-dancing leading-tight"
          >
            Skapa en magisk
            <strong
              class="block sm:text-6xl lg:text-7xl font-extrabold text-rose-400 mt-2"
            >
              Julst채mning
            </strong>
          </h1>
          <p
            class="mt-4 text-lg sm:text-xl max-w-lg mx-auto sm:mx-0"
          >
            med hj채lp av v책r breda sortiment av julpynt och dekorationer!
          </p>
          <div
            class="mt-8 flex flex-wrap justify-center sm:justify-start gap-4"
          >
            <a
              class="btn-primary px-6 py-3 rounded-full shadow-md hover:shadow-lg transition-transform transform hover:scale-105"
              data-discover="true"
              href="/products"
            >
              Visit Shop
            </a>
            <a
              class="btn-secondary px-6 py-3 rounded-full shadow-md hover:shadow-lg transition-transform transform hover:scale-105"
              data-discover="true"
              href="/about"
            >
              About Us
            </a>
          </div>
        </div>
      </div>
    </section>
    <section
      class="w-full bg-rose-400 text-white text-center flex flex-col sm:flex-row items-center justify-center py-8"
    >
      <div
        class="flex flex-col items-center sm:items-start mx-8"
      >
        <div
          class="relative"
        >
          <h1
            class="font-dancing font-bold text-4xl"
          >
            Nedr채kning till
          </h1>
          <img
            alt="Christmas Santa Claus"
            class="hidden md:block absolute -top-11 -left-12 w-20 transform scale-x-[-1] rotate-[-25deg]"
            src="/images/santaclausremovebg.png"
          />
        </div>
        <h2
          class="font-dancing font-bold text-3xl"
        >
          Juldagen
        </h2>
      </div>
      <div
        class="mt-6 sm:mt-0 mx-4 flex justify-center items-center"
      >
        <div
          class="flex flex-row gap-2 sm:gap-4 mb-4"
        >
          <div
            class="flex flex-col justify-center items-center border border-gray-300 shadow-md w-24 h-24 rounded-full bg-white"
          >
            <span
              class="text-xl font-bold text-gray-800"
            >
              0
            </span>
            <span
              class="text-sm font-medium text-gray-500"
            >
              Days
            </span>
          </div>
          <div
            class="flex flex-col justify-center items-center border border-gray-300 shadow-md w-24 h-24 rounded-full bg-white"
          >
            <span
              class="text-xl font-bold text-gray-800"
            >
              0
            </span>
            <span
              class="text-sm font-medium text-gray-500"
            >
              Hours
            </span>
          </div>
          <div
            class="flex flex-col justify-center items-center border border-gray-300 shadow-md w-24 h-24 rounded-full bg-white"
          >
            <span
              class="text-xl font-bold text-gray-800"
            >
              0
            </span>
            <span
              class="text-sm font-medium text-gray-500"
            >
              Minutes
            </span>
          </div>
          <div
            class="flex flex-col justify-center items-center border border-gray-300 shadow-md w-24 h-24 rounded-full bg-white"
          >
            <span
              class="text-xl font-bold text-gray-800"
            >
              0
            </span>
            <span
              class="text-sm font-medium text-gray-500"
            >
              Seconds
            </span>
          </div>
        </div>
      </div>
    </section>
    <section
      class="my-16 w-full px-6 sm:px-12 lg:px-20 flex flex-wrap gap-8 justify-evenly"
    >
      <article
        class="relative group"
      >
        <a
          data-discover="true"
          href="/products"
        >
          <img
            alt="Julgran framsida"
            class="w-72 xs:w-96 h-80 xs:h-[450px] rounded-lg border-4 border-black cursor-pointer transition-transform duration-300 group-hover:scale-105 group-hover:shadow-lg"
            src="/images/julgranframsida.png"
          />
        </a>
        <h3
          class="text-4xl text-black absolute bottom-4 left-3 transition-colors duration-300 group-hover:text-rose-600"
        >
          Klassiska Julgranar
        </h3>
      </article>
      <article
        class="relative group"
      >
        <a
          data-discover="true"
          href="/products"
        >
          <img
            alt="Belysning"
            class="w-72 xs:w-96 h-80 xs:h-[450px] rounded-lg border-4 border-black cursor-pointer transition-transform duration-300 group-hover:scale-105 group-hover:shadow-lg"
            src="/images/belysning.png"
          />
        </a>
        <h3
          class="text-4xl text-black absolute bottom-4 left-3 transition-colors duration-300 group-hover:text-rose-600"
        >
          Belysning
        </h3>
      </article>
    </section>
  </main>
</div>
`;
